core/domain/stats_domain.py

```diff
@@ -448,8 +485,26 @@ def from_dict(cls, state_stats_dict):
             state_stats_dict['first_hit_count_v2'],
             state_stats_dict['num_times_solution_viewed_v2'],
             state_stats_dict['num_completions_v1'],
-            state_stats_dict['num_completions_v2']
-        )
+            state_stats_dict['num_completions_v2'])
+
+    @classmethod
+    def from_frontend_dict(cls, frontend_state_stats_dict):
```

Comment:
In which contexts would this be used?

Perhaps explain why we know that the v1 stats corresponding to the frontend dict must be 0?

Reply:
For frontend dicts, I assume that the values always correspond to the most recent "stats version". This is what the `update_stats` function did (in a much more verbose way), so I've encoded that into this helper method.

Reply:
Are the frontend dicts coming from the learner or the editor view?

If the former, then they aren't state stats dicts, right?

If the latter ... that seems weird, because the editor view shouldn't be writing stats to the backend.

Reply:
It's the former, and it seems the explicit name is "AggregatedStats", see: 
https://github.com/oppia/oppia/blob/50e109d41a8a7edd18ecf9bddbe6e83d5911c50f/core/templates/domain/exploration/stats-reporting-backend-api.service.ts#L38-L45

I think StateStats is still appropriate here, however, since that's what's being "aggregated".

Reply:
Not really, right? I think they are two different concepts -- StateStats is the entirety of the stats, and the thing you're passing in from the frontend is the stats for a single session. So they shouldn't be represented by the same domain class -- instead, the latter is an incremental update to the former.

Reply:
OK, I agree. I've introduced SessionStateStats to codify this difference, WDYT?